### DISCUSSION TITLE
How to use RelationManager hooks?

### DISCUSSION THREAD

**Discussion by mrneatly** [filamentphp/filament#468] [Help]
I've discovered that the components of `Filament\Resources\RelationManager` such as `AttachRecord` or `ListRecords` are capable of triggering their hook methods after some actions - this is exactly what I need in my current project. But I can't find out how to implement these hooks properly. I've tried overriding `Filament\Resources\RelationManager::$components` property and putting my own components in it, preserving original keys of the array, but once I define a namespace of a custom component under any key of the array and visit resource edit page, I get next error:

"Unable to find component: []"

It is worth mentioning that all Filament files in my project are located in a custom directory as I use [Lucid arch](https://lucidarch.dev/) for building the app. With that being said, it may be pretty obvious that this happens because of the component not registered by the Livewire. I've tried registering my custom component using `Livewire::component()` with no luck. So how to do it properly then?

Just for any case, here is my custom component:

```php
class AttachRecord extends Filament\Resources\RelationManager\AttachRecord
{
	protected function afterAttach ()
	{
		$this->owner->update();
	}
}
```

The `Filament\Resources\RelationManager::$components` overridden property looks like this:

```php
protected static $components = [
	'attach' => ProfileKind\AttachRecord::class, // <- this is my custom component
	'create' => RelationManager\CreateRecord::class,
	'edit' => RelationManager\EditRecord::class,
	'list' => RelationManager\ListRecords::class,
];
```

Ultimately, I would like to know, how to implement relation manager component hooks and what am I doing wrong if taken approach is correct. Thanks in advance!

**Comment by danharrin** [comment]
How have you tried registering the Livewire component?

**Reply by mrneatly** [reply]
I've called a `Livewire::component()` in a `boot()` method one of my app's service providers like this:

`Livewire::component('attach-profile-kind', AttachRecord::class);`

After doing this, I am able to visit resource "edit" page, but once I select an entity for attaching, I get this:

"Unable to find component: [attach-profile-kind]"

Replacing "attach-profile-kind" with any other alias gives the same result.

**Reply by veggigit** [reply]
Could you find the solution?

**Reply by mrneatly** [reply]
I've transitioned to another package as there were too many issues with the Filament at that moment.
